src/native/mono/monodroid/monodroid-glue.cc

```diff
@@ -694,7 +694,13 @@ MonodroidRuntime::mono_runtime_init ([[maybe_unused]] JNIEnv *env, [[maybe_unuse
 
 #if defined (RELEASE)
 	if (application_config.marshal_methods_enabled) {
-		xamarin_app_init (env, get_function_pointer_at_startup);
+		if (!application_config.managed_marshal_methods_lookup_enabled) {
+			xamarin_app_init (env, get_function_pointer_at_startup);
+		} else {
+			// xamarin_app_init will be called with the actual get_function_pointer method
+			// via p/invoke from JNIEnvInit.Initialize
+			xamarin_app_init (env, get_function_pointer_placeholder);
```

Comment:
Do we need this invocation *at all*?

I would hope/expect that we could just leave it with the above branch and be done:

```cpp
if (!application_config.managed_marshal_methods_lookup_enabled) {
	xamarin_app_init (env, get_function_pointer_at_startup);
}
```

Reply:
…but `get_function_pointer_placeholder` has a nice abort message, so this likely is a good idea, in case things go horrifically wrong.

Reply:
> …but get_function_pointer_placeholder has a nice abort message, so this likely is a good idea, in case things go horrifically wrong.

My thinking was: if the method is actually called before we call the p/invoke (based on my current understanding I don't think that can actually happen, but there was a use case for `get_function_pointer_at_startup` at some point, so maybe I'm wrong), then the code in the `marshal_methods*.ll` would reach `call void %get_func_ptr(...)` which would be a null pointer and the process would crash with a segfault. Instead, this would give us a clear indication that this code is reachable and we need to figure out how to call the p/invoke earlier.